openmdao/core/system.py

```diff
@@ -1954,9 +1954,10 @@ def resolve(to_match, io_types, matches, proms):
                         if fnmatchcase(i, p):
                             break
                     else:
-                        raise RuntimeError("%s: '%s' failed to find any matches for the following "
-                                           "pattern: '%s'.%s" %
-                                           (self.msginfo, call, p, empty_group_msg))
+                        if names and p != '*':
```

Comment:
This logic will skip the error message for any pattern if names is empty.  I think the issue stated that it should only skip the error message if the pattern is '*'.

Reply:
Fixed